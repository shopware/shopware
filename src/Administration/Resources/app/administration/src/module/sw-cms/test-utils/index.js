import { mount } from '@vue/test-utils';

function runGenericCmsTest(component) {
    beforeAll(async () => {
        await setupCmsEnvironment();
    });

    it('should be a Vue.js component', async () => {
        const wrapper = mount(component);
        expect(wrapper.vm).toBeTruthy();
    });
}

function runCmsBlockRegistryTest(config) {
    beforeAll(async () => {
        await setupCmsEnvironment();
        await import(config.import);
    });

    it('registers the component in the CMS block registry', async () => {
        const componentRegistry = Shopware.Component.getComponentRegistry();
        const cmsService = Shopware.Service('cmsService');

        expect(componentRegistry.has(config.component)).toBeTruthy();
        expect(componentRegistry.has(config.preview)).toBeTruthy();

        const blockConfig = cmsService.getCmsBlockConfigByName(config.name);

        expect(blockConfig.component).toBe(config.component);
        expect(blockConfig.previewComponent).toBe(config.preview);
    });
}

function runCmsElementRegistryTest(config) {
    beforeAll(async () => {
        await setupCmsEnvironment();
        await import(config.import);
    });

    it('registers the component in the CMS block registry', async () => {
        const componentRegistry = Shopware.Component.getComponentRegistry();
        const cmsService = Shopware.Service('cmsService');

        expect(componentRegistry.has(config.component)).toBeTruthy();

        if (config.preview) {
            expect(componentRegistry.has(config.preview)).toBeTruthy();
        }

        const elementConfig = cmsService.getCmsElementConfigByName(config.name);

        expect(elementConfig.component).toBe(config.component);
        expect(elementConfig.configComponent).toBe(config.config);

        if (config.preview) {
            expect(elementConfig.previewComponent).toBe(config.preview);
        }
    });
}

async function setupCmsEnvironment() {
    await import('src/module/sw-cms/store/cms-page.store');
    await import('src/module/sw-cms/service/cms.service');
    await import('src/module/sw-cms/service/cms-element-favorites.service');
    await import('src/module/sw-cms/mixin/sw-cms-element.mixin');
    await import('src/module/sw-cms/mixin/sw-cms-state.mixin');

    Shopware.State.get('session').currentUser = {
        id: 'admin',
    };
}

/**
 * @private
 * @package buyers-experience
 */
export { runGenericCmsTest, runCmsBlockRegistryTest, runCmsElementRegistryTest, setupCmsEnvironment };
