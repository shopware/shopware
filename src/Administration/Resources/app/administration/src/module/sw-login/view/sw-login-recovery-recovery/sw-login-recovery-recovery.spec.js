/**
 * @package admin
 */
import { mount } from '@vue/test-utils';

async function createWrapper() {
    return mount(await wrapTestComponent('sw-login-recovery-recovery', { sync: true }), {
        global: {
            stubs: {
                'router-link': true,
                'sw-loader': true,
                'sw-button': true,
                'sw-icon': true,
                'sw-password-field': true,
                'sw-alert': true,
            },
            provide: {
                userRecoveryService: {
                    checkHash: () => {
                        return Promise.resolve();
                    },
                    updateUserPassword: () => {
                        return Promise.resolve();
                    },
                },
            },
        },
        props: {
            hash: '',
        },
    });
}

describe('src/module/sw-login/view/sw-login-recovery-recovery', () => {
    let wrapper;

    beforeEach(async () => {
        wrapper = await createWrapper();
    });

    it('should be a Vue.js component', () => {
        expect(wrapper.vm).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();
    });

    it('should update password successful', async () => {
        wrapper.vm.$router.push = jest.fn();
        wrapper.vm.userRecoveryService.updateUserPassword = jest.fn(() => Promise.resolve());

        await wrapper.setData({
            newPassword: 'shopware',
            newPasswordConfirm: 'shopware',
        });
        await wrapper.vm.updatePassword();

        expect(wrapper.vm.$router.push).toHaveBeenCalledWith({
            name: 'sw.login.index',
        });

        wrapper.vm.$router.push.mockRestore();
        wrapper.vm.userRecoveryService.updateUserPassword.mockRestore();
    });
});
