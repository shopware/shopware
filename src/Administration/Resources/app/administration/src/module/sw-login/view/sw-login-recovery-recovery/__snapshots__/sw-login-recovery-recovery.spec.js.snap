// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/module/sw-login/view/sw-login-recovery-recovery should be a Vue.js component 1`] = `
<div
  class="sw-login-recovery-recovery"
>
  <h2
    class="sw-login__form-headline"
  >
    <router-link-stub
      class="sw-login__back"
      to="[object Object]"
    >
      <sw-icon-stub
        class="sw-login__back-arrow"
        name="regular-long-arrow-left"
        small=""
      />
    </router-link-stub>
    
        sw-login.recovery.recovery.headline
    
  </h2>
   
  <form>
    <sw-password-field-stub
      label="sw-login.recovery.recovery.newPasswordField.label"
      value=""
    />
     
    <sw-password-field-stub
      label="sw-login.recovery.recovery.passwordConfirmField.label"
      value=""
    />
     
    <div
      class="sw-login__submit"
    >
      <sw-button-stub
        disabled="true"
        variant="primary"
      >
        
                    sw-login.recovery.recovery.submit.label
                
      </sw-button-stub>
    </div>
  </form>
</div>
`;
